stdlib/tests/collections/smt.rs

```diff
@@ -0,0 +1,124 @@
+}
+
+proptest! {
+    #[test]
+    fn arbitrary_smtget_opens_correctly(
+        ref seed in any::<[u8; 32]>()
+    ) {
+        // setup the test
+        let empty = EmptySubtreeRoots::empty_hashes(64);
+        let initial_root = Word::from(empty[0]);
+        let mut seed = *seed;
+        let key = seeded_word(&mut seed);
+        let value = seeded_word(&mut seed);
+
+        // check empty tree
+        let store = MerkleStore::new();
+        let advice_map = [];
+        let expected = [ZERO; WORD_SIZE];
+        assert_case_arbitrary_smtget_opens_correctly(
+            key,
+            expected,
+            initial_root,
+            store,
+            &advice_map
+        );
+
+        // check included leaves for all tiers
+        for depth in [16, 32, 48, 64] {
+            // compute the index and node value
+            let index = key[3].as_int();
+            let index = index >> (64 - depth);
+            let index = NodeIndex::new(depth as u8, index).unwrap();
+            let remaining = get_remaining_key(key, depth as u8);
+            let node =
+                Rpo256::merge_in_domain(&[remaining.into(), value.into()], Felt::new(depth)).into();
+
+            // setup the store and run the test
+            let mut store = MerkleStore::new();
+            let root = store.set_node(initial_root, index, node).unwrap().root;
+            let advice_map = [(node.into_bytes(), value.to_vec())];
+            assert_case_arbitrary_smtget_opens_correctly(
+                key,
+                value,
+                root,
+                store,
+                &advice_map
+            );
+        }
+
+        // check non-included leaf with same path sibling, except for last bit
+        {
+            // compute a sibling index and node value
+            let mut sibling = key[3].as_int();
+            sibling ^= 1;
+            let remaining = get_remaining_key(key, 64);
+            let sibling = NodeIndex::new(64, sibling).unwrap();
+            let node = Rpo256::merge_in_domain(&[remaining.into(), value.into()], Felt::new(64)).into();
+
+            // setup the store and run the test; expect ZERO as the leaf isn't included
+            let mut store = MerkleStore::new();
+            let root = store.set_node(initial_root, sibling, node).unwrap().root;
+            let advice_map = [(node.into_bytes(), value.to_vec())];
+            let expected = [ZERO; WORD_SIZE];
+            assert_case_arbitrary_smtget_opens_correctly(
+                key,
+                expected,
+                root,
+                store,
+                &advice_map
+            );
+        }
+    }
+}
+
+fn assert_case_arbitrary_smtget_opens_correctly(
+    key: Word,
+    value: Word,
+    root: Word,
+    store: MerkleStore,
+    advice_map: &[([u8; 32], Vec<Felt>)],
+) {
+    let source = r#"
+                use.std::collections::smt
+
+                begin
+                    exec.smt::get
+                end
+            "#;
```

Comment:
minor nit: I think there is some extra indentation here.

Reply:
Fixed!